packages/frontend/src/utils/format.ts

```diff
@@ -2,24 +2,35 @@ import { BigNumber, FixedNumber, utils } from 'ethers'
 import Network from 'src/models/Network'
 import { commafy, prettifyErrorMessage, toTokenDisplay } from '.'
 
-export function formatError(error: any, network?: Network) {
-  if (!error) {
-    return
+type PossibleError = {
+  data?: {
+    message?: string
   }
+  message?: string
+}
 
-  let errMsg = 'Something went wrong. Please try again.'
+export function formatError(error: unknown, network?: Network): string {
   if (typeof error === 'string') {
-    errMsg = error
-  } else if (error?.data?.message) {
-    errMsg = error.data.message
-  } else if (error?.message) {
-    errMsg = error.message
+    return prettifyErrorMessage(error)
```

Comment:
I think if the error parameter is a string then it should still go through the if statement checks below that format the error message, instead of returning here.